service/matching/taskListManager.go

```diff
@@ -573,6 +566,42 @@ func (c *taskListManagerImpl) shouldReload() bool {
 	return c.config.EnableTasklistIsolation() != c.enableIsolation && c.taskListKind != types.TaskListKindSticky
 }
 
+func (c *taskListManagerImpl) getIsolationGroupForTask(ctx context.Context, taskInfo *persistence.TaskInfo) string {
+	if c.isIsolationEnabled() && len(taskInfo.PartitionConfig) > 0 {
+		partitionConfig := make(map[string]string)
+		for k, v := range taskInfo.PartitionConfig {
+			partitionConfig[k] = v
+		}
+		partitionConfig[partition.WorkflowRunIDKey] = taskInfo.RunID
+		var isolationGroup string
+		group, err := c.partitioner.GetIsolationGroupByDomainID(ctx, taskInfo.DomainID, partitionConfig)
+		if err != nil {
+			c.logger.Error("Failed to get isolation group from partition library", tag.WorkflowID(taskInfo.WorkflowID), tag.WorkflowRunID(taskInfo.RunID), tag.TaskID(taskInfo.TaskID), tag.Error(err))
+		} else {
+			isolationGroup = group
+		}
+		// check if the isolation group has pollers, if not use the fallback logic
```

Comment:
I think we want to guard this fallback behaviour with a dynamic-config guard with domain filter, I can imagine there'll be scenarios were we do and don't want this behaviour.

Reply:
We have to fallback, and there is no other choice, because if we don't, async task processing is totally blocked.

Reply:
I keep forgetting 🤦